src/grt/src/GlobalRouter.cpp

```diff
@@ -545,6 +545,7 @@ void GlobalRouter::updateDirtyNets(std::vector<Net*>& dirty_nets)
       dirty_nets.push_back(db_net_map_[db_net]);
     }
   }
+  dirty_nets_.clear();
```

Comment:
@eder-matheus I do not think this original code is correct. I suspect that if we hit the return statement then the dirty_nets_.clear() call will not be made. The only reason I  made this PR was because I saw that the dirty_nets_ vector size somehow kept growing linearly with every call. 

```
    std::vector<Net*> dirty_nets;
    updateDirtyNets(dirty_nets);

    if (dirty_nets.empty()) {  
      return;  
    }
```

I will delete the original call since you also like the new place where the vector is cleared :+1:

Reply:
You're right. I didn't notice this `return` could be called even when the `dirty_nets_` vector is not empty. Thanks for the clarification!